transaction-status/src/parse_token/extension/token_metadata.rs

```diff
@@ -0,0 +1,347 @@
+    instruction: &TokenMetadataInstruction,
+    account_indexes: &[u8],
+    account_keys: &AccountKeys,
+) -> Result<ParsedInstructionEnum, ParseInstructionError> {
+    match instruction {
+        TokenMetadataInstruction::Initialize(metadata) => {
+            check_num_token_accounts(account_indexes, 4)?;
+            let Initialize { name, symbol, uri } = metadata;
+            let value = json!({
+                "metadata": account_keys[account_indexes[0] as usize].to_string(),
+                "updateAuthority": account_keys[account_indexes[1] as usize].to_string(),
+                "mint": account_keys[account_indexes[2] as usize].to_string(),
+                "mintAuthority": account_keys[account_indexes[3] as usize].to_string(),
+                "name": name,
+                "symbol": symbol,
+                "uri": uri,
+            });
+            Ok(ParsedInstructionEnum {
+                instruction_type: "initializeTokenMetadata".to_string(),
+                info: value,
+            })
+        }
+        TokenMetadataInstruction::UpdateField(update) => {
+            check_num_token_accounts(account_indexes, 2)?;
+            let UpdateField { field, value } = update;
+            let value = json!({
+                "metadata": account_keys[account_indexes[0] as usize].to_string(),
+                "updateAuthority": account_keys[account_indexes[1] as usize].to_string(),
+                "field": field,
+                "value": value,
+            });
+            Ok(ParsedInstructionEnum {
+                instruction_type: "updateTokenMetadataField".to_string(),
+                info: value,
+            })
+        }
+        TokenMetadataInstruction::RemoveKey(remove) => {
+            check_num_token_accounts(account_indexes, 2)?;
+            let RemoveKey { key, idempotent } = remove;
+            let value = json!({
+                "metadata": account_keys[account_indexes[0] as usize].to_string(),
+                "updateAuthority": account_keys[account_indexes[1] as usize].to_string(),
+                "key": key,
+                "idempotent": *idempotent,
+            });
+            Ok(ParsedInstructionEnum {
+                instruction_type: "removeTokenMetadataKey".to_string(),
+                info: value,
+            })
+        }
+        TokenMetadataInstruction::UpdateAuthority(update) => {
+            check_num_token_accounts(account_indexes, 2)?;
+            let UpdateAuthority { new_authority } = update;
+            let mut value = json!({
+                "metadata": account_keys[account_indexes[0] as usize].to_string(),
+                "updateAuthority": account_keys[account_indexes[1] as usize].to_string(),
+            });
+            let map = value.as_object_mut().unwrap();
+            if let Some(new_authority) = Option::<Pubkey>::from(*new_authority) {
+                map.insert("newAuthority".to_string(), json!(new_authority.to_string()));
+            }
```

Comment:
Same as earlier, can you just have it be `null` if nothing is set?
```suggestion
            let value = json!({
                "metadata": account_keys[account_indexes[0] as usize].to_string(),
                "updateAuthority": account_keys[account_indexes[1] as usize].to_string(),
                "newAuthority": Option::<Pubkey>::from(*new_authority).map(|v| v.to_string()),
            });
```

Reply:
Yeah sure. Are you cool with this approach for `TokenMetadataInstruction::Emit`?

Reply:
Lmk. I've pushed the other requested changes.

Reply:
Yeah that made sense to me, if there's *only* a start, you kind of get that there's no end, and vice versa